src/superagent-promise-plugin.js

```diff
@@ -3,8 +3,7 @@ function endPromise(req) {
 
   return new _Promise(function (resolve, reject) {
     req.end(function (err, res) {
-      err = err || res.error;
-      if (err) reject(err);
+      if (err || res.error) reject(res);
```

Comment:
Hmmm. maybe something like this will cover all cases:

```js
err = err || res.error;
if (err) {
  err.response = res;
  reject(err);
} else {
  resolve(res);
}
```